lib/cocoapods-core/specification/dsl.rb

```diff
@@ -666,6 +666,12 @@ def dependency(*args)
         attributes_hash['dependencies'][name] = version_requirements
       end
 
+      def dependency=(args)
+        joined = args.join('\', \'')
+        arguments = "\'#{joined}\'"
+        raise Informative, "Cannot assign value to `dependency`. Did you mean: `dependency #{arguments}`?"
```

Comment:
thing we could just do `dependency #{args.map(&:inspect).join(', ')`

Reply:
oh cool. didn't know that existed

Reply:
this results in escaped quotes: `dependency \"JSONKit\", \"1.5\"`